worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+        await super(HadesContext, self).shutdown()
+
+    # ----------------- Client start up and ending section end  --------------------------------
+
+    # ----------------- Package Management section starts --------------------------------
+
+    def on_package(self, cmd: str, args: dict):
+        # This is what is done when a package arrives.
+        if cmd == "Connected":
+            # What should be done in a connection package
+            self.cache_items_received_names.clear()
+            self.missing_locations_cache = args["missing_locations"]
+            self.checked_locations_cache = args["checked_locations"]
+            self.hades_slot_data = args["slot_data"]
+            if not (self.hades_slot_data["version_check"] == self.polycosmos_version):
+                stringError = "WORLD GENERATED WITH POLYCOSMOS " + self.hades_slot_data["version_check"] \
+                            + " AND CLIENT USING POLYCOSMOS " + self.polycosmos_version + "\n"
+                stringError += "THIS ARE NOT COMPATIBLE"
+                raise Exception(stringError)
+            self.location_name_to_id = self.get_location_name_to_id()
+            if "death_link" in self.hades_slot_data and self.hades_slot_data["death_link"]:
+                asyncio.create_task(self.update_death_link(True))
+                self.deathlink_enabled = True
+            self.is_connected = True
+            self.is_receiving_items_from_connect_package = True 
+
+        if cmd == "RoomInfo":
+            # What should be done when room info is sent.
+            self.seed_name = args["seed_name"]
+        
+        if cmd == "RoomUpdate":
+            if "checked_lodations" in args:
+                collect_locations_cache = ""
+                for location in args["checked_locations"]:
+                    collect_locations_cache += self.location_names.lookup_in_slot(location) + "-"
+                if len(collect_locations_cache) > 0:
+                    collect_locations_cache = collect_locations_cache[:-1]
+                    subsume.Send(styx_scribe_send_prefix + "Locations collected:" + collect_locations_cache)
+
+        if cmd == "ReceivedItems":
+            # What should be done when an Item is recieved.
+            # NOTE THIS GETS ALL ITEMS THAT HAVE BEEN RECIEVED! WE USE THIS FOR RESYNCS!
+            for item in args["items"]:
+                self.cache_items_received_names += [self.item_names.lookup_in_slot(item.item)]
+
+            # We ignore sending the package to hades if just connected,
+            # since the game is not ready for it (and will request it itself later)
+            if self.is_receiving_items_from_connect_package:
+                return
+            self.send_items()
+
+        if cmd == "LocationInfo":
+            if self.creating_location_to_item_mapping:
+                self.creating_location_to_item_mapping = False
+                asyncio.create_task(self.create_location_to_item_dictionary(args["locations"]))
```

Comment:
This would be another instance of a dropped task, but `create_location_to_item_dictionary` doesn't even need to be `async`, so you can drop that and just call it normally.